src/Approximations/iterative_refinement.jl

```diff
@@ -72,17 +74,31 @@ function approximate(S::LazySet, ɛ::Float64)::Array{Approximation2D, 1}
     push!(queue, Approximation2D(pn, DIR_NORTH, pw, DIR_WEST))
     push!(queue, Approximation2D(pw, DIR_WEST, ps, DIR_SOUTH))
     push!(queue, Approximation2D(ps, DIR_SOUTH, pe, DIR_EAST))
+
     i = 1
-    # iterative refinement
+    N = length(queue)
     while i <= length(queue)
         if (queue[i].err <= ɛ)
             i += 1
-        else
 
+        else
             (la1, la2) = refine(S, queue[i])
-            queue[i] = la1
-            insert!(queue, i+1, la2)
+
+            if (maximum(map(abs, la1.q - la2.q)) < TOL_POINTS)
+                # update the constraints if a redundance is detected
```

Comment:
Can you comment this code?
I do not understand it.
For instance, is it correct that you never need `la2`?